Sources/ImagePipeline.swift

```diff
@@ -575,8 +576,13 @@ public /* final */ class ImagePipeline {
         }
 
         // Store data in data cache (in case it's enabled))
-        if !data.isEmpty, let dataCache = configuration.dataCache, let key = session.request.urlString {
-            dataCache[key] = data
+        if let dataCache = configuration.dataCache, let key = session.request.urlString {
+            if data.isEmpty {
+                dataCache.removeData(for: key)
```

Comment:
I don't think that you should remove data at this point. If you get to this point in the pipeline it means that the data wasn't found in the `dataCache` and was fetched from the network instead.

Reply:
You're right. In addition, the data removal logic must be in the cache mechanism itself. The changes have been made.